src/honeybadger.cr

```diff
@@ -147,4 +149,10 @@ module Honeybadger
   def self.notify(exception : Exception) : Nil
     Dispatch.send_async Payload.new(exception)
   end
+
+  def self.notify(exception : Exception, context : ContextHash) : Nil
```

Comment:
Does this allow keyword arguments like in the example in #4? We will probably want to add additional options in the future (and this matches the Ruby API).

```crystal
Honeybadger.notify(exception, {
  context: {
    user_id: 1
  }
})
```

Reply:
I've declared ContextHash above as: `alias ContextHash = Hash(String, String | Int32)`. Crystal syntax can be a bit picky about the difference between these, which all look like basically the same thing to a rubyist:

- `a = { one: "one" }` where `a` is a NamedTuple (like a fixed hash)
- `a = { :one => "one" }` where `a` is a Hash(Symbol, String)
- `a = { "one" => "one" }` where `a` is a Hash(String, String)

Crystal provides named parameters implicitly on methods, so the calling syntax would be:

```crystal
Honeybadger.notify(exception, context: { user_id: 1 })
```

The type as defined would restrict sending values which are not Int32's or Strings -- I might recommend tightening that up to be just strings rather than attempting to implement anything and everything as a hash value.